### PR TITLE
Feat - user profile

### PR DESCRIPTION
**Pull request by SinchanaMS**
This request adds the following features:

1. User Profile
2. Edit Profile Modal - to update Cover Pic, Profile Pic, Bio and Portfolio
3. Lists the User's Posts

### Preview:
1. User Profile
image

2. Edit Profile Modal
image

3. User's posts
image

### CODE REVIEW

**Review by srejitk — APPROVED** [review]
Good readable code with well defined firebase calls. Would suggest you segregate the API calls to different actions for better readability. Good work!

File: src/pages/profile/UserProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        const docSnap = await getDoc(docRef);
+        setUserData(docSnap.data());
+      };
+      if (auth.currentUser) userDetails();
+    },
+    //eslint-disable-next-line
+    [auth.currentUser]
+  );
+
+  console.log(user);
```
Comment (Adarsh-Patel-Dev) [inline src/pages/profile/UserProfile.jsx]: remove log here

**Review by Adarsh-Patel-Dev — APPROVED** [review]
The code is clean and properly structured
UI is awesome as always
You can merge after working on suggestions given by Srejit's

**Comment by SinchanaMS** [comment]
> Good readable code with well defined firebase calls. Would suggest you segregate the API calls to different actions for better readability. Good work!

Yes, I do plan on segregating the functions when I refactor the code. Thank you for the review!

**Comment by SinchanaMS** [comment]
Thank you for the reviews. Merging to dev.

**Pull request merged**